File: ios/Sources/BeaconRadio/IdentityConfirmationUploading.swift
import Foundation

/// Abstrae la subida a `participants/{device_id_hash}` del campo
/// `identidad_verificada_caso_a` (Firebase Auth + Firestore en la
/// implementación real, `FirebaseIdentityConfirmationUploader` en la capa
/// de app) para que `IdentityConfirmationUploader` sea testeable sin tocar
/// red — mismo molde que `MeshStateUploading`/`ParticipantUploading`.
public protocol IdentityConfirmationUploading {
    func upload(deviceIdHash: Data, completion: @escaping (Result<Void, Error>) -> Void)
}

File: android/beaconradio/src/main/kotlin/com/farosos/beaconradio/IdentityConfirmationUploading.kt
package com.farosos.beaconradio

/**
 * Abstrae la subida del campo `identidad_verificada_caso_a` a
 * `participants/{device_id_hash}` (Firebase Auth + Firestore en la
 * implementación real, `FirebaseIdentityConfirmationUploader` en la capa de
 * app) para que `IdentityConfirmationUploader` sea testeable mockeando este
 * primitivo nativo en vez de sustituir la clase entera.
 */
interface IdentityConfirmationUploading {
    fun upload(deviceIdHash: ByteArray, onResult: (Result<Unit>) -> Unit)
}
